wallet/src/lib.rs

```diff
@@ -57,6 +60,89 @@ impl WalletState {
         Self { utxos, pk_index }
     }
 
+    pub fn fund_tx<G: GasConstants>(
+        &self,
+        tx_builder: MantleTxBuilder,
+        change_pk: PublicKey,
+        pks: impl IntoIterator<Item = PublicKey>,
+    ) -> Result<MantleTxBuilder, WalletError> {
+        let mut utxos: Vec<Utxo> = pks
+            .into_iter()
+            .filter_map(|pk| self.pk_index.get(&pk))
+            .flatten()
+            .map(|id| self.utxos[id])
+            .collect();
```

Comment:
Should we consume the PublicKeys? or `&PublicKey` is enough?

Reply:
I figured since PublicKey is Copy, it's low overhead to consume here.